sphere-models/src/it/java/io/sphere/sdk/products/ProductCategoryOrderHintTest.java

```diff
@@ -0,0 +1,168 @@
+package io.sphere.sdk.products;
+
+import io.sphere.sdk.categories.CategoryFixtures;
+import io.sphere.sdk.categories.CategoryOrderHints;
+import io.sphere.sdk.products.commands.ProductCreateCommand;
+import io.sphere.sdk.products.queries.ProductProjectionQuery;
+import io.sphere.sdk.products.search.ProductProjectionSearch;
+import io.sphere.sdk.producttypes.ProductTypeFixtures;
+import io.sphere.sdk.search.SearchSort;
+import io.sphere.sdk.test.IntegrationTest;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static io.sphere.sdk.products.ProductProjectionComparators.comparingCategoryOrderHints;
+import static io.sphere.sdk.test.SphereTestUtils.*;
+import static io.sphere.sdk.utils.MapUtils.mapOf;
+import static java.util.Comparator.comparing;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ProductCategoryOrderHintTest extends IntegrationTest {
```

Comment:
For a quick view, I'd draw here:
```
Product | Cat 1  |  Cat 2
1           | 0,1      | 0,89
2           | 0,3      | 0,88
3           | 0,2      | -
4           | -          | -
5           | 0,4      | 0,86
```
Just because I had to do it to have a clear view :P